### PR TITLE
Finishing up the editors

### PR DESCRIPTION
**Pull request by zsarnett**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/editor/config-elements/hui-generic-entity-row-editor.ts [hunk 1]
```diff
@@ -45,127 +38,107 @@ export class HuiGenericEntityRowEditor
...
-        this._config = {
-          ...this._config,
-          [target.configValue!]: value,
-        };
-      }
+  private _computeLabelCallback = (schema: HaFormSchema) => {
+    if (schema.name === "entity") {
+      return `${this.hass!.localize(
+        "ui.panel.lovelace.editor.card.generic.entity"
+      )} (${this.hass!.localize(
```
Comment (balloob) [inline src/panels/lovelace/editor/config-elements/hui-generic-entity-row-editor.ts:128]: I don't think that we should add required to the label like this. I think that selectors just need to render an `*` if something is required. (and so we do that by setting `required: true`,  not in label computation)

File: src/panels/lovelace/editor/config-elements/hui-humidifier-card-editor.ts [hunk 2]
```diff
@@ -37,81 +48,39 @@ export class HuiHumidifierCardEditor
...
-        delete this._config[target.configValue!];
-      } else {
-        this._config = { ...this._config, [target.configValue!]: target.value };
-      }
+  private _computeLabelCallback = (schema: HaFormSchema) => {
+    if (schema.name === "entity") {
+      return `${this.hass!.localize(
+        "ui.panel.lovelace.editor.card.generic.entity"
+      )} (${this.hass!.localize(
+        "ui.panel.lovelace.editor.card.config.required"
```
Comment (balloob) [inline src/panels/lovelace/editor/config-elements/hui-humidifier-card-editor.ts:76]: Same.

File: src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts [hunk 3]
```diff
@@ -19,6 +18,18 @@ const cardConfigStruct = assign(
...
 );
 
+const SCHEMA: HaFormSchema[] = [
+  { name: "title", selector: { text: {} } },
+  {
+    name: "",
+    type: "grid",
+    schema: [
+      { name: "url", required: true, selector: { text: {} } },
+      { name: "aspect_ratio", selector: { text: {} } },
```
Comment (balloob) [inline src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts:28]: Do we have a placeholder for the aspect ratio that we use when it's not filled in ? That also helps the user to show what kind of values we expect.

Comment (balloob) [reply src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts:28]: And maybe one day we make a visual aspect ratio selector.

File: src/panels/lovelace/editor/config-elements/hui-light-card-editor.ts [hunk 4]
```diff
@@ -190,9 +168,35 @@ export class HuiLightCardEditor
...
+  private _valueChanged(ev: CustomEvent): void {
+    fireEvent(this, "config-changed", { config: ev.detail.value });
   }
+
+  private _computeLabelCallback = (schema: HaFormSchema) => {
+    if (schema.name === "entity") {
+      return `${this.hass!.localize(
+        "ui.panel.lovelace.editor.card.generic.entity"
+      )} (${this.hass!.localize(
+        "ui.panel.lovelace.editor.card.config.required"
```
Comment (balloob) [inline src/panels/lovelace/editor/config-elements/hui-light-card-editor.ts:180]: same

File: src/panels/lovelace/editor/config-elements/hui-logbook-card-editor.ts [hunk 5]
```diff
@@ -41,118 +49,58 @@ export class HuiLogbookCardEditor
...
+        .data=${this._config}
+        .schema=${SCHEMA}
+        .computeLabel=${this._computeLabelCallback}
+        @value-changed=${this._valueChanged}
+      ></ha-form>
+      <h3>
+        ${`${this.hass!.localize(
+          "ui.panel.lovelace.editor.card.generic.entities"
+        )} (${this.hass!.localize(
+          "ui.panel.lovelace.editor.card.config.required"
```
Comment (balloob) [inline src/panels/lovelace/editor/config-elements/hui-logbook-card-editor.ts:78]: same

File: src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts [hunk 6]
```diff
@@ -86,140 +85,29 @@ export class HuiPictureEntityCardEditor
...
-          </div>
-        </div>
-        <hui-theme-select-editor
-          .hass=${this.hass}
-          .value=${this._theme}
-          .configValue=${"theme"}
-          @value-changed=${this._valueChanged}
-        ></hui-theme-select-editor>
         <div class="side-by-side">
           <hui-action-editor
```
Comment (balloob) [inline src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts:112]: Ugh, these shouldn't even be in the editor.

Comment (balloob) [reply src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts:112]: (for a different PR)

File: src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts [hunk 7]
```diff
@@ -279,23 +153,38 @@ export class HuiPictureEntityCardEditor
...
 
-  static get styles(): CSSResultGroup {
-    return configElementStyle;
-  }
+  private _computeLabelCallback = (schema: HaFormSchema) => {
+    if (schema.name === "entity") {
+      return `${this.hass!.localize(
+        "ui.panel.lovelace.editor.card.generic.entity"
+      )} (${this.hass!.localize(
+        "ui.panel.lovelace.editor.card.config.required"
```
Comment (balloob) [inline src/panels/lovelace/editor/config-elements/hui-picture-entity-card-editor.ts:173]: Same

**Review by balloob — DISMISSED** [review]

**Review by balloob — APPROVED** [review]

File: src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts (hunk 3 above)
Comment (zsarnett) [reply src/panels/lovelace/editor/config-elements/hui-iframe-card-editor.ts:28]: The default is 50%. Which is shown to the user in the editor

**Pull request merged**